### PR TITLE
release: sourcegraph@3.22.0

### PR DESCRIPTION
**Pull request by pecigonzalo**
This pull request is part of the Sourcegraph 3.22.0 release.

* [Release campaign](https://k8s.sgdev.org/organizations/sourcegraph/campaigns/release-sourcegraph-3.22.0)
* [Tracking issue](https://github.com/sourcegraph/sourcegraph/issues/15023

### :warning: Additional changes required

- [x] Update the upgrade guides in `doc/admin/updates`
- [x] Ensure all other pull requests in the campaign has been merged before merging this pull request

cc @pecigonzalo

### CODE REVIEW

File: doc/admin/updates/docker_compose.md [hunk 1]
```diff
@@ -11,6 +11,12 @@ Upgrades should happen across consecutive minor versions of Sourcegraph. For exa
 
 <!-- GENERATE UPGRADE GUIDE ON RELEASE (release tooling uses this to add entries) -->
 
+## 3.21 -> 3.22
+
+No manual migration required.
+
+Please upgrade to the [`v3.22.0` tag of deploy-sourcegraph-docker](https://github.com/sourcegraph/deploy-sourcegraph-docker/tree/v3.22.0/docker-compose) by following the [standard upgrade procedure](#standard-upgrade-procedure).
```
Comment (bobheadxi) [inline doc/admin/updates/docker_compose.md:18]: minor: should we just point to the release branch for these links? Ideally patches should never require manual migration right?

**Review by bobheadxi — APPROVED** [review]

**Comment by bobheadxi** [comment]
```
  handler_test.go:62: Could not find Docker Compose or Pure Docker release 3.22.0 on GitHub. Response code 404 Not Found from https://github.com/sourcegraph/deploy-sourcegraph-docker/releases/tag/v3.22.0, err: <nil>
```

Did not realize we had a test for this - previously, the tag would happen before this merge, and I tried to move this to after 😢

File: doc/admin/updates/docker_compose.md (hunk 1 above)
Comment (pecigonzalo) [reply doc/admin/updates/docker_compose.md:18]: I theory you are correct, but I believe there is not harm to point to the exact version, and its less risky.

**Comment by pecigonzalo** [comment]
@bobheadxi Maybe I should create the tag now, or would that break the CLI?

File: doc/admin/updates/docker_compose.md (hunk 1 above)
Comment (bobheadxi) [reply doc/admin/updates/docker_compose.md:18]: > there is not harm to point to the exact version

Someone could come along to these docs, and we might forget to update the specific tag, and pull an old unpatched version with bugs :(

**Pull request merged**